common/scripts/bump-beta-version.js

```diff
@@ -13,12 +13,19 @@ const main = () => {
 
 const bumpBetaVersion = (currentVersion) => {
   const versionStrs = currentVersion.split('-beta.');
-  const newBeta = versionStrs[1] === undefined ? 0 : Number.parseInt(versionStrs[1]) + 1;
-  const [major, minor, patch] = versionStrs[0].split('.');
-  const newPatch = Number.parseInt(patch) + 1;
+  const [major, minor] = versionStrs[0].split('.');
+  const betaVersion = versionStrs[1];
+  
+  // If there is no beta version then we simply append '-beta.0'
+  if (betaVersion === undefined) {
+    return [versionStrs, 0].join('-beta.');
+  }
 
-  // We will bump the patch version when create a beta on top of a stable one
-  return [[major, minor, newBeta === 0 ? newPatch : patch].join('.'), `${newBeta}`].join('-beta.');
+  const newBeta = Number.parseInt(betaVersion) + 1;
+  // We will bump the minor version when we create a beta on top of a stable one
+  const newMinor = Number.parseInt(betaVersion) === 0 ? Number.parseInt(minor) + 1 : minor;
+
+  return [[major, newMinor, 0].join('.'), `${newBeta}`].join('-beta.');
```

Comment:
Here we are mixing string templates and string joins, to make this simpler and much easier to parse when reading, just use a string template:
```ts
return `${major}.${minor}.0-beta.${newBeta}`;
```